### PR TITLE
Ethereum contract: Lottery

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: golem/ethereum/contracts/Lottery.sol [hunk 1]
```diff
@@ -8,13 +8,6 @@ contract Lottery {
...
     }
 
-    function payout() {
-        if (msg.sender == owner) {
-            if (msg.sender.send(ownerDeposit))
-                ownerDeposit = 0;
-        }
-    }
-
     // in real-life this struct can fill 2 storage words
```
Comment (badb) [inline golem/ethereum/contracts/Lottery.sol]: Is it still true?

Comment (chfast) [inline golem/ethereum/contracts/Lottery.sol]: Yes. In stage 1 we have: value (32b), payer (20b) and maturity (should fit 8b). In stage 2 random value replaces maturity and payer.

**Comment by badb** [comment]
Looks great, but we still should consider possibility of sending `check` for someone else and receiving a percantage of the prize.

**Comment by chfast** [comment]
"Someone else" would need to know the proof. So only the lottery participants can do it. Let's consider that later on.

**Pull request merged**